### ISSUE TITLE
v1.2 DEX initContainers missing resource definition

### ISSUE THREAD

**Issue by pbmoses** [redhat-developer/gitops-operator#183]
**Describe the bug**

**To Reproduce**
1. upgrade to 1.2
2. allow the Dex rs to run
3. pods fail
4. Error creating: pods "openshift-gitops-dex-server-..." is forbidden: failed quota: openshift-gitops-compute-resources: must specify cpu,limits.cpu,limits.memory,memory

**Expected behavior**
Resources need to be defined for initcontainers

**Screenshots**

**incorrect:**
    initContainers:
        - name: copyutil
          image: >-
            registry.redhat.io/openshift-gitops-1/argocd-rhel8@sha256:df4b11a78ab8f8a3ee758a1a2b549c190ce1467cdd098a1c10468bbf6e76a596
          command:
            - cp
            - '-n'
            - /usr/local/bin/argocd
            - /shared/argocd-dex
          resources: {}

**working:** 

  spec:
      restartPolicy: Always
      initContainers:
        - name: copyutil
          image: >-
            registry.redhat.io/openshift-gitops-1/argocd-rhel8@sha256:3e6521a3610b23dce99f4eb643171ac9172808e86f3ca4154f5d548a286bb95f
          command:
            - cp
            - '-n'
            - /usr/local/bin/argocd
            - /shared/argocd-dex
          resources:
            limits:
              cpu: 500m
              memory: 256Mi
            requests:
              cpu: 250m
              memory: 128Mi
**Additional context**
<img width="782" alt="Screen Shot 2021-07-29 at 3 58 48 PM" src="https://user-images.githubusercontent.com/24865328/127576067-7c97e07c-1db5-4b25-b31a-3a62ed0bef52.png">
<img width="560" alt="Screen Shot 2021-07-29 at 3 59 33 PM" src="https://user-images.githubusercontent.com/24865328/127576070-5c6fb74f-346f-4269-ac86-d01940204b45.png">

**Comment by iam-veeramalla** [comment]
Hi @pbmoses , Thanks for raising issue with us.

Did you add resource requests/limits manually to fix the issue? Looking at the code, upgrade for Dex should work as expected.
https://github.com/argoproj-labs/argocd-operator/blob/c5a3e0d1496f01a4786b81e478d796ff1e970c5b/pkg/controller/argocd/deployment.go#L418

However, the easy workaround for this issue is to remove the resourcequota on the `openshift-gitops` namespace using the below command.

`oc delete resourcequota openshift-gitops-compute-resources -n openshift-gitops`

**Comment by pbmoses** [comment]
I did fix this manually by adding resources limits to the initContainers. As you mentioned, removing the resourcequota on the namespace would also work.
I do believe, based on what I saw, that without manual intervention, the existing dex-server pods would remain while the new deployment/replicaset would fail (as well as future  which would create an issue unless manually resolved. )

**
[failing-deployment-openshift-gitops-dex-server.txt](https://github.com/redhat-developer/gitops-operator/files/6907444/failing-deployment-openshift-gitops-dex-server.txt)

    - type: ReplicaFailure
      status: 'True'
      lastTransitionTime: '2021-07-30T12:08:57Z'
      reason: FailedCreate
      message: >-
        pods "openshift-gitops-dex-server-58985bc68f-jl28w" is forbidden: failed
        quota: openshift-gitops-compute-resources: must specify
        cpu,limits.cpu,limits.memory,memory
